client/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala

```diff
@@ -791,11 +796,25 @@ case class DeltaSharingSource(
       add.id -> add.url
     }.toMap
 
+    // For streaming queries, we return a DataFrame.
+    // The FileIndex here is a one-time use object to list files.
     val params = new RemoteDeltaFileIndexParams(
-      spark, initSnapshot, deltaLog.client.getProfileProvider)
+      spark, initSnapshot, deltaLog.client.getProfileProvider, Some(queryParamsHashId))
```

Comment:
This actually have a strong assumption (which should be true), but not enforced in code: `queryParamsHashId` will NOT change after the rpc is made, and before the files are listed.